eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -198,6 +198,26 @@ trait Service extends Logging {
                           completeRpcFuture(req.id, (paymentHandler ? ReceivePayment(Some(MilliSatoshi(amountMsat.toLong)), description)).mapTo[PaymentRequest].map(PaymentRequest.write))
                         case _ => reject(UnknownParamsRejection(req.id, "[description] or [amount, description]"))
                       }
+                      
+                      case "checkinvoice" => req.params match {
+                        case JString(paymentRequest) :: Nil  => Try(PaymentRequest.read(paymentRequest)) match {
+                          case Success(pr) => {
+                            completeRpc(req.id,pr)
+                          }
+                            // setting the payment amount
+                          case _ => reject(RpcValidationRejection(req.id, s"payment request is not valid"))
+                        }
+                      }
```

Comment:
We should reject the method if the `paymentRequest` argument is missing, with something like:

`case _ => reject(UnknownParamsRejection(req.id, "[payment_request]"))`

Reply:
fixed